### PR TITLE
Bug: Reversed layer ordering

### LINKED ISSUE #189: Bug: Layer ordering can end up in a reversed state
A bug surfaced that causes the property order to be set incorrectly. In normal operation, this bug was resolved by a double negative, that is, a situation where the ordered array was reversed, and then reversed again later.

However, the health check endpoint calls a getter which mutates the ordered array in place, causing this bug to occur.

The list of active sources is reversed by the [`Properties.sources()`](https://github.com/rapid7/propsd/blob/v1.3.1/lib/properties.js#L34) getter. That getter is called by the [`Sources.health()`](https://github.com/rapid7/propsd/blob/v1.3.1/lib/sources.js#L155) function used by both the `/v1/health` and `/v1/status` APIs.

Additionally, the list of layers is reversed every time [`Properties.build()`](https://github.com/rapid7/propsd/blob/v1.3.1/lib/properties.js#L117) is called.

### PR DESCRIPTION
**Pull request by dgreene-r7**
This PR refactors the `Properties` class to never reverse internal arrays. This fixes problems where the `Properties#sources` getter and `Properties#build()` were mutating objects in place. It reverses the order in which Consul, EC2 Metadata, and static properties are loaded in `server.js`.

It also pins the node version to the latest LTS/argon release and reorders the way the Consul, Metadata, and local Config layers are loaded.

Resolves #189 and supersedes #192.

### CODE REVIEW

**Comment by fmitchell-r7** [comment]
@dgreene-r7, would you mind splitting this into two PRs, one for the bug fix with tests and one for the eslint and doc improvements? I feel like it makes writing the changelog for releases easier when I can say, "Here's a PR that fixes a bug" and "Here's a PR that improves documentation".

**Comment by dgreene-r7** [comment]
Can do!

File: lib/properties.js [hunk 1]
```diff
@@ -29,9 +32,8 @@ class Properties extends EventEmitter {
       // Get Dynamic layers' source instances
       .concat(this.layers.map((layer) => layer.source).filter((source) => !!source))
 
-      // Add the active View's sources. Reversed for precedence consistency with Layers:
-      // Entry 0 is most precedent, LAST is least precedent.
-      .concat(this.active.sources.reverse());
+      // Add the active View's sources.
+      .concat(this.active.sources);
```
Comment (jmanero-r7) [inline lib/properties.js]: I had to stare at this for a while to make sure...

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-4.4.0
+lts/argon
```
Comment (dgreene-r7) [inline .nvmrc]: Crap, you're right. I'm splitting the stuff out into 2 separate PRs so I'll drop this change.

File: lib/properties.js (hunk 1 above)
Comment (dgreene-r7) [inline lib/properties.js]: I feel like this is diminishing returns. The only use of this getter is in the health check and the order doesn't change anything beyond which order the source statuses are displayed in the `/v1/status` output.

Comment (jmanero-r7) [inline lib/properties.js]: Yep. _shrugs_ Keeping the API unchanged is basically what I'm getting at.

**Pull request closed without merging**